src/analysis/normed_space/weak_dual.lean

```diff
@@ -182,3 +190,255 @@ begin
+  rw [hφ, ← hg₀, weak_dual.to_Pi],
+  dsimp,
+  simp only [algebra.id.smul_eq_mul, continuous_linear_map.map_add, add_sub_cancel,
+             sub_self, continuous_linear_map.map_smul, add_sub_cancel', sub_self],
+end
+
+/-- Elements of the closure of the range of the embedding
+`weak_dual.to_Pi : weak_dual 𝕜 E → (E → 𝕜)` can be viewed as linear maps `E → 𝕜`. -/
+def linear_of_mem_closure_range
+  (f : (E → 𝕜)) (hf : f ∈ closure (range (weak_dual.to_Pi 𝕜 E))) :
+  E →ₗ[𝕜] 𝕜 :=
+{ to_fun := f,
+  map_add' := begin
+    intros z₁ z₂,
+    have key := linear_of_mem_closure_range' f hf (1 : 𝕜) (1 : 𝕜) z₁ z₂,
+    rwa [one_smul, one_smul, one_smul 𝕜 _, one_smul 𝕜 _] at key,
+  end,
+  map_smul' := begin
+    intros c z,
+    have key := linear_of_mem_closure_range' f hf c (0 : 𝕜) z (0 : E),
+    rwa [zero_smul, zero_smul, add_zero, add_zero] at key,
+  end, }
+
+lemma linear_of_mem_closure_range_apply
+  (f : E → 𝕜) (hf : f ∈ closure (range (weak_dual.to_Pi 𝕜 E))) (z : E) :
+  linear_of_mem_closure_range f hf z = f z := rfl
+
+/-- Elements of the closure of the image under `weak_dual.to_Pi : weak_dual 𝕜 E → (E → 𝕜)` of
+a subset defined by a non-strict bound on the norm still satisfy the same bound. -/
+lemma norm_eval_le_of_mem_closure_norm_eval_le
+  (z : E) (c : ℝ) (f : E → 𝕜)
+  (hf : f ∈ closure ((weak_dual.to_Pi 𝕜 E) '' {x' : weak_dual 𝕜 E | ∥ x' z ∥ ≤ c})) :
+  ∥ f z ∥ ≤ c :=
+begin
+  suffices : ∀ (ε : ℝ), 0 < ε → ∥ f (z) ∥ ≤ c + ε,
+  { exact le_of_forall_pos_le_add this, },
+  intros ε ε_pos,
+  have nhd := mem_nhds_Pi_normed_field f z ε_pos,
+  cases mem_closure_iff_nhds.mp hf _ nhd with g hg,
+  simp only [mem_image, mem_inter_eq, mem_set_of_eq] at hg,
+  rcases hg with ⟨tri, ⟨y', ⟨at_z_le, eq_g⟩⟩⟩,
+  have eq : weak_dual.to_Pi 𝕜 E y' z = y' z := rfl,
+  rw [← eq_g, eq] at tri,
+  have key := norm_add_le_of_le tri.le at_z_le,
+  rwa [sub_add_cancel, add_comm] at key,
+end
+
+/-- Elements of the closure of the image under `weak_dual.to_Pi : weak_dual 𝕜 E → (E → 𝕜)` of
+a polar `polar s` of a neighborhood `s` of the origin are continuous (linear) functions. -/
+lemma continuous_of_mem_closure_polar
+  {s : set E} (s_nhd : s ∈ 𝓝 (0 : E)) (φ : E → 𝕜)
+  (hφ : φ ∈ closure ((weak_dual.to_Pi 𝕜 E) '' (weak_dual.polar 𝕜 s))) :
+  continuous φ :=
+begin
+  cases polar_bounded_of_nhds_zero 𝕜 s_nhd with c hc,
+  have c_nn : 0 ≤ c := le_trans (norm_nonneg _) (hc 0 (zero_mem_polar 𝕜 s)),
+  have hφ' : φ ∈ closure (range (weak_dual.to_Pi 𝕜 E)),
+  { apply mem_of_mem_of_subset hφ _,
+    apply closure_mono,
+    simp only [preimage_univ, preimage_range, subset_univ, image_subset_iff], },
+  set flin := linear_of_mem_closure_range φ hφ' with hflin,
+  suffices : continuous flin,
+  { assumption, },
+  apply linear_map.continuous_of_bound flin c,
+  intros z,
+  set θ := λ (ψ : E → 𝕜), ∥ ψ z ∥ with hθ,
+  have θ_cont : continuous θ,
+  { apply continuous.comp continuous_norm,
+    exact continuous_apply z, },
```

Comment:
```suggestion
  from (continuous_apply z).norm,
```

Reply:
I like this one-line-proof, and I don't see why it shouldn't work, but it doesn't. In this and variants, I get errors of the form:

```
type mismatch at application
  (continuous_apply z).norm
term
  continuous_apply z
has type
  continuous (λ (p : Π (i : E), ?m_1 i), p z)
but is expected to have type
  continuous ?m_5
```

I revert back to the two-line-proof for now.